include/lsst/afw/math/PixelScaleBoundedField.h

```diff
@@ -0,0 +1,103 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+/** \file
+ * A BoundedField that gives the amount a pixel is distorted at each point.
+ */
+#ifndef LSST_AFW_MATH_PixelScaleBoundedField_h_INCLUDED
+#define LSST_AFW_MATH_PixelScaleBoundedField_h_INCLUDED
+
+#include "lsst/afw/geom/SkyWcs.h"
+#include "lsst/afw/math/BoundedField.h"
+#include "lsst/pex/exceptions.h"
+
+namespace lsst {
+namespace afw {
+namespace math {
+
+/**
+ * A BoundedField that gives the amount a pixel is distorted at each point.
+ *
+ * This is ratio of the SkyWcs-transformed pixel area to the transformed pixel area at the SkyWcs center, or
+ * equivalently the determinant of the Jacobian of the SkyWcs Transform.
+ *
+ * Typically used to move an image or source flux between surface brightness and fluence space.
+ */
+class PixelScaleBoundedField : public table::io::PersistableFacade<PixelScaleBoundedField>,
+                               public BoundedField {
+public:
+    /**
+     *  Create a PixelScaleBoundedField from a bounding box and SkyWcs.
+     */
+    PixelScaleBoundedField(lsst::geom::Box2I const &bbox, geom::SkyWcs const &skyWcs)
+            : BoundedField(bbox),
+              _skyWcs(skyWcs),
+              _inverseScale(1.0 / std::pow(skyWcs.getPixelScale().asDegrees(), 2)) {}
+
+    ~PixelScaleBoundedField() override = default;
+
+    PixelScaleBoundedField(PixelScaleBoundedField const &) = default;
+    PixelScaleBoundedField(PixelScaleBoundedField &&) = default;
+    PixelScaleBoundedField &operator=(PixelScaleBoundedField const &) = delete;
+    PixelScaleBoundedField &operator=(PixelScaleBoundedField &&) = delete;
+
+    /// Get the contained SkyWcs
+    geom::SkyWcs const &getSkyWcs() const { return _skyWcs; }
+    // Get the cached inverse pixel scale
+    double getDefaultInverseScale() const { return _inverseScale; }
+
+    /// @copydoc BoundedField::evaluate
+    double evaluate(lsst::geom::Point2D const &position) const override;
+
+    /// TransformBoundedField is not persistable.
+    bool isPersistable() const noexcept override { return false; }
+
+    /// @copydoc BoundedField::operator*
+    std::shared_ptr<BoundedField> operator*(double const scale) const override {
+        throw LSST_EXCEPT(pex::exceptions::LogicError, "Not implemented");
+    }
+
+    /// @copydoc BoundedField::operator==
+    bool operator==(BoundedField const &rhs) const override;
+
+    // protected:
+    // std::string getPersistenceName() const override;
+
+    // std::string getPythonModule() const override;
+
+    // void write(OutputArchiveHandle &handle) const override;
+
+private:
+    geom::SkyWcs const _skyWcs;
+    // Inverse pixel scale (square degrees), for convenience.
+    double const _inverseScale;
+
+    std::string toString() const override { return ""; }
```

Comment:
Right, I was trying to think of what to put in there, but I think I'll just leave it unimplemented, since I can't come up with a good `str` representation.

Reply:
Should at least return the class name, and the WCS pointer address might be useful for debugging if we can't come up with anything better.

Reply:
The updated version does that.